docs/design/rubygems.md

```diff
@@ -0,0 +1,528 @@
+
+```ruby
+# Plain Ruby Gem
+"https://rubygems.org/gems/#{name}-#{version}.gem"
+
+# Platform-specific Gem
+"https://rubygems.org/gems/#{name}-#{version}-#{platform}.gem"
+```
+
+This means that we can easily just download the plain Ruby Gems, and skip all platform-specific Gems altogether. From
+the testing I did, this did not seem to affect the ability to perform a hermetic build in any way, Bundler just
+compiled everything from source.
+
+To avoid confusing users if a corner case shows up, we can add a `WARNING` log in case extra platforms are detected,
+and also document it properly. 
+
+Proper support for pre-compiled binaries should be probably left as a follow-up feature, similarly to what was done
+with [pip wheels](https://github.com/containerbuildsystem/cachi2/blob/main/docs/pip.md#distribution-formats).
+
+#### Checksum verification
+Since checksums in the `Gemlock.file` is still a feature in development (see [checksums](#dependency-checksums)), we
+can postpone implementing support for it until the feature is delivered.
+
+We need to decide if we will report all dependencies as having missing checksums in the SBOM, or not.
+
+#### Dev dependencies
+Bundler declares all dev dependencies under the `:development`
+[group](#dependency-groups-or-how-bundler-deals-with-dev-dependencies). Unfortunately, groups declared in the `Gemfile`
+are not reflected in the `Gemfile.lock`.
+
+To implement proper reporting of dev dependencies, we'll very likely need to also parse the `Gemfile`. It can be done
+as a follow-up if the need arises.
+
+#### Prefetching Bundler
+When running `bundle install`, Bundler will always try to fetch the exact version that is pinned in the `Gemfile.lock`
+to perform the install. When doing an offline install from cache, a warning message is instead printed, but Bundler
+will usually perform the install as expected.
+
+To allow users to use the pinned version instead of only relying on the Bundler version present in the base image,
+Cachi2 could also prefetch the specific Bundler version needed for that project. This is easy to achieve, since Bundler
+is treated as an ordinary Gem: https://rubygems.org/gems/bundler. 
+
+This feature, however, is out of scope for the initial implementation, and could be added if there's user demand for
+it.
+
+### Providing the content for the hermetic build
+
+#### Setting the Bundler configuration
+
+The order of precedence for Bundler configuration options is as follows:
+
+1. Local config (`<project_root>/.bundle/config or $BUNDLE_APP_CONFIG/config`)
+2. Environment variables (ENV)
+3. Global config (`~/.bundle/config`)
+4. Bundler default config
+
+Since the local configuration takes higher precedence than the environment variables (except `BUNDLE_APP_CONFIG`), we
+need to set the Bundler configuration options to make the build work.
+
+In order to do this, we can either use `inject-files` to overwrite the `.bundle/config` directory in the source folder,
```

Comment:
how would `inject-files` know how to overwrite each config based on our build config JSON? We'd have to create mappings of config paths -> env vars (or directly config keys) and re-format those as key=value pairs in the bundler config files in question. I guess there's no way around it, is it? Although we don't seem to handle this in Yarn, a bug probably?

Reply:
> Although we don't seem to handle this in Yarn, a bug probably?

Correction, we do! Because config opts have the correct order of precedence, so 1 set of env vars works like a charm.